### PR TITLE
Detector-Competition-New: add IP2Location api key detector

### PR DESCRIPTION
**Pull request by fumblehool**
### Description:
Adds Ip2Location Detector for Issue #1726 

### Checklist:
* [X] Tests passing (`make test-community`)?
* [X] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### Verification:
Screenshot 2023-10-19 at 12 39 54 AM

### How to Test?
* Goto `https://www.ip2location.io/` and Sign-up.
* Once You're logged in, you can get API access token from the dashboard.
Screenshot 2023-10-19 at 9 52 19 PM

### CODE REVIEW

File: pkg/detectors/ip2location/ip2location.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+			}
+			res, err := client.Do(req)
+			if err == nil {
+				defer res.Body.Close()
+				if res.StatusCode >= 200 && res.StatusCode < 300 {
+					s1.Verified = true
+				} else if res.StatusCode == 401 {
+					// The secret is determinately not verified (nothing to do)
+				} else {
+					if detectors.IsKnownFalsePositive(resMatch, detectors.DefaultFalsePositives, true) {
```
Comment (trufflesteeeve) [inline pkg/detectors/ip2location/ip2location.go:64]: Can we move the `detectors.IsKnownFalsePositive` check outside of the `if verify {` block, so we always check for false positives, even when verification is turned off?

An example can be found in the [twilio detector here](https://github.com/trufflesecurity/trufflehog/blob/main/pkg/detectors/twilio/twilio.go#L90).

**Review by trufflesteeeve — COMMENTED** [review]
Hi @fumblehool , thank you for your contribution. I have a few PR comments. Could you also add instructions on how to create an IP2Location api key in a comment or your PR description?

**Comment by fumblehool** [comment]
Hi @trufflesteeeve , thanks for the review 😄 
I've updated the code as per the comments, please have a look and let me know if any other change is required.
Thanks 👍

**Comment by zricethezav** [comment]
@fumblehool this PR LGTM, just need to resolve merge conflicts

**Comment by fumblehool** [comment]
@zricethezav Done, please merge this.
Once this this merge, I'll rebase the other PR to resolve the merge conflict there.

**Comment by fumblehool** [comment]
@zricethezav this is ready to be merged.
Please merge this. Once this is merged, I'll rebase other PRs

**Review by zricethezav — APPROVED** [review]

**Pull request merged**